### PR TITLE
Productionize Soccer Stream Detection

### PR DESCRIPTION
**Pull request by cyberj0g**
**What does this pull request do? Explain your changes. (required)**
Make content detection functionality production-ready.

**Limitations**
Current implementation de facto uses a separate ad-hoc data stream to output results - the webhook. Detection results are not included with the renditions, because the DNN filter is run in a separate output, which doesn't produce any renditions. Model output is included into AVFrame metadata only for the purpose of transferring data back to LPMS. This would work for internal process of unwanted content detection, but the process needs to be re-worked to produce 'output stream' to end user, if required in the future for ML capabilities.

**Specific updates (required)**
* Add `-detectContent` command line flag to enable content detection for transcoder
* Fix list of T capabilities to include SceneDetection capability, if enabled
* Create correct list of T capabilities in case capabilities testing is disabled

**How did you test each of these updates (required)**
- Stdout echo web server for content detection webhook
```
while true; do {   echo -ne "HTTP/1.0 200 OK\r\nContent-Length: 0"; } | nc -l -p 8081 -q 1 ; done
```
- B
```
livepeer -broadcaster -rtmpAddr :1936 -cliAddr 127.0.0.1:7937 -nvidia all -httpAddr 127.0.0.1:8936 -orchAddr 127.0.0.1:8935 -orchSecret 11111111111 -v 6 -detectionWebhookUrl http://localhost:8081/
```
- OT
```
livepeer -transcoder -orchestrator -serviceAddr 127.0.0.1:8935 -nvidia all -v 7 -detectContent
```
- Stream
```
ffmpeg -re -i bbb.mp4 -c:a copy -c:v copy -f hls -hls_time 2 -headers 'Livepeer-Transcode-Configuration:
                                                                    {"detection": {"freq": 2, "sampleRate": 10, "sceneClassification": [{"name": "soccer"}]}}' http://localhost:8936/live/movie3/
```
- Webhook output
```
{"manifestID":"movie3","seqNo":10,"sceneClassification":[{"name":"soccer","probability":0.00676799938082695}]}
```

**Does this pull request close any open issues?**

#2394 

**Checklist:**

- [X] Read the [contribution guide](./doc/contributing.md)
- [X] `make` runs successfully
- [X] All tests in `./test.sh` pass
- [n/a] README and other documentation updated
- [X] [Pending changelog](./CHANGELOG_PENDING.md) updated

**TBD: more tests for standalone T**

### CODE REVIEW

**Comment by cyberj0g** [comment]
Converting this to draft, because discovered the following while testing:
1. Doesn't work in a separate O-T configuration, because detection results are not transmitted between T and O by HTTP
2. Detection webhook, which I expect to be the primary way of detecting unwanted content, [not get called](https://github.com/livepeer/go-livepeer/blob/eb7e829b79b7dd50b6bbd0dcf83fa06f0fd845f3/server/broadcast.go#L992) when verification is enabled.

**Review by thomshutt — APPROVED** [review]
Looks good, do we have any metrics around this at the moment? It'd be good when we begin testing to be able to see how many detections we're getting

**Comment by cyberj0g** [comment]
> Looks good, do we have any metrics around this at the moment? It'd be good when we begin testing to be able to see how many detections we're getting

No, there are no metrics there. I expect that classification results will be reported through webhook, and metrics probably should be logged on that end.

**Review by thomshutt — APPROVED** [review]

File: common/util_test.go [hunk 1]
```diff
@@ -458,15 +458,6 @@ func TestParseAccelDevices_WrongDriver(t *testing.T) {
 	getPCI = originGetPCI
 }
 
-func TestParseAccelDevices_FailedDetection(t *testing.T) {
```
Comment (cyberj0g) [inline common/util_test.go:461]: @red-0ne what is the purpose of this test? It always fails, if Nvidia GPU is present.

Comment (red-0ne) [reply common/util_test.go:461]: You are right. I think it should setup its `GetGPU`/`GetCPU`
The idea was to fail when no GPUs or appropriate PCI devices are detected (it fails now if you have a real one)
A minimal resolution would be
```go
	getGPU = func() ([]*gpu.GraphicsCard, error) {
		return []*gpu.GraphicsCard{}, nil
	}
	getPCI = func() ([]*pci.Device, error) {
		return []*pci.Device{}, nil
	}
```

**Pull request merged**